orca-bakery/src/main/groovy/com/netflix/spinnaker/orca/bakery/tasks/manifests/BakeManifestContext.java

```diff
@@ -39,24 +39,26 @@
   // Jackson can use to deserialize.
   public BakeManifestContext(
       @Nullable @JsonProperty("inputArtifacts")
-          List<CreateBakeManifestTask.InputArtifactPair> inputArtifacts,
+          List<CreateBakeManifestTask.InputArtifact> inputArtifacts,
       @JsonProperty("expectedArtifacts") List<ExpectedArtifact> expectedArtifacts,
       @JsonProperty("overrides") Map<String, Object> overrides,
       @JsonProperty("evaluateOverrideExpressions") Boolean evaluateOverrideExpressions,
       @JsonProperty("templateRenderer") String templateRenderer,
       @JsonProperty("outputName") String outputName,
       @JsonProperty("namespace") String namespace,
-      @Nullable @JsonProperty("inputArtifact")
-          CreateBakeManifestTask.InputArtifactPair inputArtifact,
+      @Nullable @JsonProperty("inputArtifact") CreateBakeManifestTask.InputArtifact inputArtifact,
       @JsonProperty("rawOverrides") Boolean rawOverrides) {
-    this.inputArtifacts = inputArtifacts;
+    this.inputArtifacts = inputArtifacts == null ? new ArrayList<>() : inputArtifacts;
```

Comment:
Maybe slightly more idiomatic as `this.inputArtifacts = Optional.of(inputArtifacts).orElse(new Arraylist<>())`.  But this is also completely fine too.

Reply:
That's way better, updated!